platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/DefaultValueSnapshotter.java

```diff
@@ -157,6 +157,11 @@ public ValueSnapshot array(ImmutableList<ValueSnapshot> elements, Class<?> array
             return new ArrayValueSnapshot(elements);
         }
 
+        @Override
+        public ValueSnapshot primitiveArray(Object value) {
```

Comment:
❓ Could we follow the same pattern as for other ValueSnapshots for arrays, e.g. we would have `IsolatedArrayOfPrimitiveValueSnapshot` and implement methods there?

Reply:
Do you mean for consistency or are there other implications here?

Reply:
Mostly for consistency. But also that way you would be forced to implement `appendToHasher` and `asSnapshot` that now throw exception.

Reply:
Yeah, I was wondering about `appendToHasher` and `asSnapshot`. That should also will give us the opportunity to make the `byte[]` case more efficient. I'll do that, thanks.